### ISSUE TITLE
Elementor generated CSS file, permissions not right. (0600)

### ISSUE THREAD

**Issue by numanzahid** [elementor/elementor#9488]
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

I am using elementor since one year now. Problem I am continually facing is elementor CSS file print does not work properly with external CSS files. It generates the correct. It makes changes to the file. but the file permission is set to 0600. the result is, page looks fine with elementor editor, but when you go to the front-end the page is broken. and no css is leaded.
also there is ERR_ABORTED 403 and in some cases 403 Forbidden error with css in console.
Now I can simply go to the /wp-content/uploads/elementor/css and change the permission. the page will look fine. but the problem reapers when I create the new page. I will have to go to the file again to change the permission.
and If I click Regenerate files in tools. all the files are set to 0600 again.
I can simply use internal embedding, but this is not a proper solution.
I have seen this problem with almost 50 new WordPress installs, as I do work for the a web-developer company and we use elementor on almost every website.

I have mostly worked with LSWS, I dont know if problem is same with other servers.

## Steps to reproduce

on new wordpress install leave settings to default. make sure css print method is set to external file.
make new page.
publish the page.
edit with elementor and save
page is broken on the front end.
there is 403 error in console about css files. 

## Isolating the problem
- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment
<details>
<summary>
https://pastebin.com/FPgU335e
</summary>
</details>

**Comment by shilo-ey** [comment]
Hi @numanzahid 

From what I can tell it's an issue with LiteSpeed servers.

Please contact our support at support @elementor.com for guidance in resolving this issue and **mention that you were directed by me and reference this issue**.

In addition, if you could provide cPanel credentials for better debugging it would be helpful.

thanks!

**Comment by shilo-ey** [comment]
Hi @numanzahid 

Thanks for providing the required credentials.

After investigating this issue, there is a problem that origins from LiteSpeed servers configuration, which restricts (permission 0600) the option to view CSS files for users who are not the owners of the generated file (e.g - not the server)

Elementor didn't change the way it writes files nor the way it handles them. 

From other users reports, we can see that this issue might be related to `cagefs` module configuration. [ref.](https://wordpress.org/support/topic/css-permission-0600/)

I suggest you to contact your hosting provider and explain the issue.

If you had a website that was functioning properly before updating Elementor, and after the update "broke", please contact our support and mention that you were directed by me and reference this issue.
Please make sure to provide cPanel credentials for better debugging.


For now, as you mentioned, you can change the output method of styles to "Inline".

Update this thread if the hosting provider solved it for you.

Thanks.

**Comment by numanzahid** [comment]
Thank for the help @shilo-ey . I am in contact with email support team. plus I am going to create ticket with my hosting provider as well. 

One thing for sure, it has nothing to do with updates. I am facing this problem since very long now. I have updated multiple installations. 

I will let you know when the issue is resolved. 

thanks again for the guide.

**Comment by numanzahid** [comment]
@shilo-ey 

thanks for the help. I contacted my hosting provider. I told them the details about cageFS error and linked the thread. Thankfully they understood what was the issue. and they fixed it, and now elementor is working fine with external css print. I did same with my clients hosting and my other hosting account. Elementor external css print is working same everywhere after the fix. So this solution is confirmed working.

For hosting solutions provider or to create ticked with host, this information might help.

`The CSS files created by WordPress or one of its plugins - Elementor, for all WordPress installations are generated with 0600 permissions rather then 0644 is due to an Imunify360 update. Imunify360 causing umask and permission issues in CloudLinux servers hence Umask of new files created in WordPress and CMS files getting wrong. We have fixed the issue by remounting the cageFS.`

So the key is to simply **remount the cageFS**

Hope this helps.

happy building blocks.
